velox/functions/lib/Re2Functions.cpp

```diff
@@ -805,71 +880,95 @@ std::vector<std::shared_ptr<exec::FunctionSignature>> re2ExtractSignatures() {
   };
 }
 
-std::pair<PatternKind, vector_size_t> determinePatternKind(StringView pattern) {
-  vector_size_t patternLength = pattern.size();
-  vector_size_t i = 0;
-  // Index of the first % or _ character.
-  vector_size_t wildcardStart = -1;
-  // Index of the first character that is not % and not _.
-  vector_size_t fixedPatternStart = -1;
-  // Total number of % characters.
-  vector_size_t anyCharacterWildcardCount = 0;
-  // Total number of _ characters.
-  vector_size_t singleCharacterWildcardCount = 0;
-  auto patternStr = pattern.data();
+PatternParameters determinePatternKind(const StringView& pattern) {
+  auto i = 0;
+  const auto patternLength = pattern.size();
+  const auto patternString = pattern.str();
+  auto wildcardStart = 0;
+  auto wildcardCount = 0;
```

Comment:
Use num instead of count in the naming so numWildcard, numFixedPattern, numAnyWildcard, numSingleWildcard